plugins/catalog-backend/src/processing/DefaultCatalogProcessingOrchestrator.ts

```diff
@@ -183,20 +187,47 @@ export class DefaultCatalogProcessingOrchestrator
 
     for (const processor of this.options.processors) {
       if (processor.preProcessEntity) {
-        try {
-          res = await processor.preProcessEntity(
-            res,
-            context.location,
-            context.collector.forProcessor(processor),
-            context.originLocation,
-            context.cache.forProcessor(processor),
-          );
-        } catch (e) {
-          throw new InputError(
-            `Processor ${processor.constructor.name} threw an error while preprocessing`,
-            e,
-          );
-        }
+        let innerRes = res;
+        res = await tracer.startActiveSpan(
+          'catalog_pre_process_entity',
```

Comment:
Would it make sense to have a hierarchy of

```
ProcessingRun      { entityRef=component:default/foo }
  ProcessingStage  { entityRef=component:default/foo stage=preProcessEntity }
    ProcessingStep { entityRef=component:default/foo stage=preProcessEntity processorName=MyProcessor1 }
    ProcessingStep { entityRef=component:default/foo stage=preProcessEntity processorName=MyProcessor2 }
  ProcessingStage  { entityRef=component:default/foo stage=postProcessEntity }
    ProcessingStep { entityRef=component:default/foo stage=postProcessEntity processorName=MyProcessor1 }
```

Or something like that? Makes it easy to see which stage is the expensive one, while still being able to drill down and/or filter by processor name

Reply:
Sure, works for me. Naming is hard.. and pretty arbitrary for span names

Reply:
I've made this change

Reply:
Stages are now in too :)